quay_search.py

```diff
@@ -0,0 +1,135 @@
+                corrector = searcher.corrector("title")
+                alternative_strings =  corrector.suggest(p_search_string, limit=2)
+
+                # get all repositories with suggested keywords
+                for i in alternative_strings:
+                    search_string = u"*"
+                    search_string += i
+                    search_string += "*"
+                    query = QueryParser("title", self._index.schema).parse(search_string)
+                    results_tmp = searcher.search(query)
+                    results.extend(results_tmp)
+           
+            # get all versions for the found tools
+            dict_results = {}
+            for i in results:
+                additional_info = self.get_additional_repository_information(i['title'])
+                versions = []
+                for j in additional_info:
+                    versions.append(j)
+                dict_results[i['title']] = versions
+            
+            print "The query ", '\033[1m' + p_search_string + '\033[0m', " resulted in", len(results) ,"result(s).",
+            
+            if p_non_strict:
+                print "The search lists the results for ",
+                for i in xrange(len(alternative_strings)):
+                    print '\033[1m' + alternative_strings[i] + '\033[0m', ", ",
+                print "too."
+            for i in dict_results:
+                print i, "\n\t\t\t", 
+                for j in dict_results[i]:
+                    print j, "\t\t", "docker pull quay.io/"+self._organization+"/"+i+":"+j+"\n\t\t\t",
+                print "\n"
+   
+    def get_additional_repository_information(self, p_repository_string):
+        """Function downloads additional information from quay.io to get the tag-filed which includes the version number."""
+        url = 'https://quay.io/api/v1/repository/'
+        url += self._organization
+        url += '/'
+        url += p_repository_string
+        r = requests.get(url, headers={'Accept-encoding': 'gzip'}, timeout=12)
+
+        json_decoder = json.JSONDecoder()
+        decoded_request = json_decoder.decode(r.text)
+        return decoded_request['tags']
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Searches in a given quay organization for a repository')
+    parser.add_argument('--organization', dest='organization_string', type=str,
+                        help='Change organization. Default is mulled.')
+    parser.add_argument('--non-strict', dest='non_strict_bool', action="store_true",
+                        help='Autocorrection of typos activated. Lists more results but can be confusing.\
+                        For too many queries quay.io blocks the request and the results can be incomplete.')
+    parser.add_argument('search', type=str,
+                        help='The name of the tool you want to search for.')        
+    args = parser.parse_args()
+    
+    if args.organization_string is None:
```

Comment:
`argparse` contains a native implementation for a default value, this avoids confusion.